x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -177,7 +281,28 @@ export async function getAllAgentsByKuery(
   agents: Agent[];
   total: number;
 }> {
-  const res = await getAgentsByKuery(esClient, { ...options, page: 1, perPage: SO_SEARCH_LIMIT });
+  const pitId = await openAgentsPointInTime(esClient);
+
+  const res = await getAgentsByKueryPit(esClient, {
+    ...options,
+    page: 1,
+    perPage: SO_SEARCH_LIMIT,
+    pitId,
+  });
+
+  while (res.agents.length < res.total) {
```

Comment:
Could we define a separate variable for collecting the agents instead of modifying res.agents?